src/Symfony/Bundle/FrameworkBundle/Debug/CliRequest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Symfony\Bundle\FrameworkBundle\Console\TraceableCommand;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @internal
+ */
+class CliRequest extends Request
+{
+    public function __construct(public readonly TraceableCommand $command)
+    {
+        parent::__construct(
+            attributes: ['_controller' => \get_class($command->command), '_virtual' => 'command'],
+            server: $_SERVER,
+        );
+    }
+
+    // Methods below allow to populate a profile, thus enable search and filtering
+    public function getUri(): string
+    {
+        if ($this->server->has('SYMFONY_CLI_BINARY_NAME')) {
+            $binary = $this->server->get('SYMFONY_CLI_BINARY_NAME').' console';
+        } else {
+            $binary = $this->server->get('argv')[0];
+        }
+
+        return $binary.' '.$this->command->input;
+    }
+
+    public function getMethod(): string
+    {
+        return $this->command->isInteractive ? 'INTERACTIVE' : 'BATCH';
+    }
+
+    public function getResponse(): Response
+    {
+        return new class($this->command->exitCode) extends Response {
+            public function __construct(private readonly int $exitCode)
+            {
+                parent::__construct();
+            }
+
+            public function getStatusCode(): int
+            {
+                return $this->exitCode;
+            }
+        };
+    }
+
+    public function getClientIp(): string
+    {
+        $application = $this->command->getApplication();
+
+        return $application->getName().' '.$application->getVersion();
```

Comment:
This is also very hacky, maybe could we hardcode `127.0.0.1` and provide another method to get the app name and version? Or store this data in attributes?

Reply:
Same here, this is a way to fill a `Profile` with useful data for the default listing of profiles on the main profiler page where we have no access to requests nor collectors.